http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/WsmApiClientProvider.scala

```diff
@@ -86,67 +97,93 @@ class HttpWsmClientProvider[F[_]](baseWorkspaceManagerUrl: Uri)(implicit F: Asyn
     attempt <- F.delay(wsmApi.getAzureDatabase(workspaceId.value, wsmResourceId.value)).attempt
-    state = attempt match {
-      case Right(result) => Some(result.getMetadata.getState.getValue)
+    db = attempt match {
+      case Right(result) => Some(result.getMetadata)
       case Left(_)       => None
     }
-  } yield toWsmStatus(state)
+  } yield db
 
-  override def getNamespaceState(token: String, workspaceId: WorkspaceId, wsmResourceId: WsmControlledResourceId)(
-    implicit
+  override def getNamespace(token: String, workspaceId: WorkspaceId, wsmResourceId: WsmControlledResourceId)(implicit
     ev: Ask[F, AppContext],
     log: StructuredLogger[F]
-  ): F[WsmState] = for {
+  ): F[Option[ResourceMetadata]] = for {
     wsmApi <- getControlledAzureResourceApi(token)
     attempt <- F.delay(wsmApi.getAzureKubernetesNamespace(workspaceId.value, wsmResourceId.value)).attempt
-    state = attempt match {
-      case Right(result) => Some(result.getMetadata.getState.getValue)
+    namespace = attempt match {
+      case Right(result) => Some(result.getMetadata)
       case Left(_)       => None
     }
-  } yield toWsmStatus(state)
+  } yield namespace
 
-  override def getIdentityState(token: String, workspaceId: WorkspaceId, wsmResourceId: WsmControlledResourceId)(
-    implicit
+  override def getIdentity(token: String, workspaceId: WorkspaceId, wsmResourceId: WsmControlledResourceId)(implicit
     ev: Ask[F, AppContext],
     log: StructuredLogger[F]
-  ): F[WsmState] = for {
+  ): F[Option[ResourceMetadata]] = for {
     wsmApi <- getControlledAzureResourceApi(token)
     attempt <- F.delay(wsmApi.getAzureManagedIdentity(workspaceId.value, wsmResourceId.value)).attempt
-    state = attempt match {
-      case Right(result) => Some(result.getMetadata.getState.getValue)
+    id = attempt match {
+      case Right(result) => Some(result.getMetadata)
       case Left(_)       => None
     }
+  } yield id
+
+  override def getWsmState(token: String,
+                           workspaceId: WorkspaceId,
+                           wsmResourceId: WsmControlledResourceId,
+                           resourceType: WsmResourceType
+  )(implicit
+    ev: Ask[F, AppContext],
+    log: StructuredLogger[F]
+  ): F[WsmState] = for {
+    resource <- resourceType match {
+      case WsmResourceType.AzureVm =>
+        getVm(token, workspaceId, wsmResourceId)
+      case WsmResourceType.AzureDatabase =>
+        getDatabase(token, workspaceId, wsmResourceId)
+      case WsmResourceType.AzureKubernetesNamespace =>
+        getNamespace(token, workspaceId, wsmResourceId)
+      case WsmResourceType.AzureManagedIdentity =>
+        getIdentity(token, workspaceId, wsmResourceId)
+      case WsmResourceType.AzureDisk =>
+        getDisk(token, workspaceId, wsmResourceId)
+      case WsmResourceType.AzureStorageContainer =>
+        F.pure(None) // TODO: no get endpoint for a storage container in WSM yet
```

Comment:
Do you know if there is an existing ticket in the WSM backlog for that? If so it might be worth linking here to the comment